added a tutorial - installin cstrike server (#506)
hetzneronline/community-content: tutorials/installing-cs-server/01.en.md

---
SPDX-License-Identifier: MIT
path: "/tutorials/installing-cs-server"
slug: "installing-cs-server"
date: "2022-10-31"
title: "Installing and configuring a dedicated Counter-Strike Server"
short_description: "Installing, configuring and running an official instance of a server for CS 1.6."
tags: ["cs", "counter-strike", "server", "gaming"]
author: "Filip Furár"
author_link: "https://github.com/numero69"
author_img: "https://avatars.githubusercontent.com/u/26823803?v=4"
author_description: "Just an enthusiast."
language: "en"
available_languages: ["en"]
header_img: "header-x"
cta: "cloud"
---

## Introduction

In this tutorial I will be showing you how to successfully run a Counter-Strike 1.6 dedicated server and also teach you how to set it up. You will end up with a fully working server you and your friends can connect to, and also any one playing the game. This tutorial will guide you through the whole process without any useless steps or inormation that might confuse you and is aimed torwards anyone, be it a beginner or professional who just never tried starting a CS server.

> 

Reviewer: ## Comment

`app_update 90 validate` did not work for me at all. However, according to the [dedicated servers list](https://developer.valvesoftware.com/wiki/Dedicated_Servers_List), the installation should theoretically be possible with anonymous login as you said. And it worked for you, right? I think there should be a warning at the beginning of the tutorial that installation might fail even after several attempts.

## Suggestion

```suggestion

> Note: Due to a bug, the installation could fail even after several attempts. However, according to the [dedicated servers list](https://developer.valvesoftware.com/wiki/Dedicated_Servers_List), the installation should work with "anonymous login", it could just take a while.

```

Now, please SSH into your server and make sure everything is fine before proceeding.

Valve recommends running steamcmd from a newly created user for security purpouses. However, this step can be ignored. If you would like to create a new user called `steam`, please run these commands:

```
sudo useradd -m steam
sudo passwd steam
```
You will be prompted to create a password. You can then navigate via `cd` command into its `/home` directory.
```
sudo -u steam -s
cd /home/steam
```

As our OS is 64bit, we will need to add the multiverse repo.

```
sudo add-apt-repository multiverse
sudo apt install software-properties-common
sudo dpkg --add-architecture i386
sudo apt update
```
```
sudo apt install lib32gcc-s1 steamcmd 
```

After executing these commands into your shell, you will have installed SteamCMD.
Execute it by running `steamcmd`

## Step 2 - `<Log into SteamCMD and configure it>`

Now, login via `login anonymous`.

Choose a directory where you want to install your server, I decided to choose `/root/cs/`

```

> force_install_dir /some/path/

Reviewer: ## Comment

`force_install_dir /some/path/` didn't work for me when I was logged in and I got this warning:
```
Please use force_install_dir before logon!
```

I had to logout, execute the command, and log back in with `login anonymous`.

## Suggestion

## Step 2 - Log into SteamCMD and configure it

Choose a directory where you want to install your server, I decided to choose `/root/cs/`

```
force_install_dir /some/path/
```

Now, login via `login anonymous` and execute the following:

```
app_update 90
app_set_config 90 mod cstrike
```

and finally,

```
app_update 90 validate
```

app_update 90
app_set_config 90 mod cstrike
```

and finally,

`app_update 90 validate`

> You might see an error on your screen. This bug is mentioned on official Valve dedicated server list, but remains unfixed to this day. Just run the command again and again, untill it install successfully.

Reviewer: ## Comment

I would suggest to add the source here for reference.

## Suggestion

```suggestion
You might see an error on your screen. This bug is mentioned in the [official Valve dedicated server list](https://developer.valvesoftware.com/wiki/Dedicated_Servers_List), but remains unfixed to this day. Just run the command again and again, until it installs successfully.
```

> ![Error installing app 90](1.png)

Reviewer: ## Comment

Please move your images to a subdirectory named `images`.

## Suggestion

```suggestion
![Error installing app 90](images/1.png)
```

Successful install should look like this.

> ![Validated successfully.](2.png)

Reviewer: ## Comment

Please move your images to a subdirectory named `images`.

## Suggestion

```suggestion
![Validated successfully.](images/1.png)
```
